test/renamer_test.dart

```diff
@@ -0,0 +1,174 @@
+        expect(renamer.rename('oldx'), isNull);
+        expect(renamer.rename('xold'), isNull);
+        expect(renamer.rename('new'), isNull);
+      });
+
+      test('space syntax with `to` clause and default key', () {
+        var renamer = Renamer.simple('old to new');
+        expect(renamer.rename('old'), equals('new'));
+        expect(renamer.rename('oldx'), isNull);
+        expect(renamer.rename('xold'), isNull);
+        expect(renamer.rename('new'), isNull);
+      });
+
+      test('space syntax with `->` clause', () {
+        var renamer = Renamer.simple('s old -> new');
+        expect(renamer.rename('old'), equals('new'));
+        expect(renamer.rename('oldx'), isNull);
+        expect(renamer.rename('xold'), isNull);
+        expect(renamer.rename('new'), isNull);
+      });
+
+      test('space syntax with `->` clause and default key', () {
+        var renamer = Renamer.simple('old -> new');
+        expect(renamer.rename('old'), equals('new'));
+        expect(renamer.rename('oldx'), isNull);
+        expect(renamer.rename('xold'), isNull);
+        expect(renamer.rename('new'), isNull);
+      });
+    });
+
+    group('backreferences', () {
+      test(r'to entire match', () {
+        var renamer = Renamer.simple(r'.+ to prefix-\0');
+        expect(renamer.rename('a'), equals('prefix-a'));
+        expect(renamer.rename('abc'), equals('prefix-abc'));
+      });
+
+      test(r'to group', () {
+        var renamer = Renamer.simple(r'/(.+)-suffix/\1/');
+        expect(renamer.rename('a-suffix'), equals('a'));
+        expect(renamer.rename('abc-suffix'), equals('abc'));
+        expect(renamer.rename('abc'), isNull);
+        expect(renamer.rename('-suffix'), isNull);
+      });
+    });
+
+    group('escaping', () {
+      test(r'spaces', () {
+        var renamer = Renamer.simple(r'a\ b to x\ y');
+        expect(renamer.rename('a b'), equals('x y'));
+      });
+
+      test(r'normal delimiters', () {
+        var renamer = Renamer.simple(r'/a\/b/x\/y/');
+        expect(renamer.rename('a/b'), equals('x/y'));
+      });
+
+      test(r'semicolons', () {
+        var renamer = Renamer.simple(r'a\;b to x\;y');
+        expect(renamer.rename('a;b'), equals('x;y'));
+      });
+    });
+
+    group('multiple keys', () {
+      test('sed-style syntax', () {
+        var renamer =
+            Renamer.map(r'url=.*/(\w+)/lib/mixins=\1=', ['namespace', 'url']);
+        expect(
+            renamer.rename(
+                {'namespace': 'mixins', 'url': 'path/button/lib/mixins'}),
+            equals('button'));
+      });
+
+      test('space syntax', () {
+        var renamer =
+            Renamer.map(r'url .*/(\w+)/lib/mixins to \1', ['namespace', 'url']);
+        expect(
+            renamer.rename(
+                {'namespace': 'mixins', 'url': 'path/button/lib/mixins'}),
+            equals('button'));
+      });
+    });
+  });
+
+  group('multiple statements', () {
```

Comment:
Test empty initial statements. In particular, it's likely that users writing shell scripts will write empty initial newlines.

Reply:
Done. The updated parser should handle any amount of spaces, newlines, and semicolons at any point before or after a statement.